tests/007-server-streams.cpp

```diff
@@ -0,0 +1,103 @@
+#include <catch2/catch_test_macros.hpp>
+#include <quic.hpp>
+#include <thread>
+
+namespace oxen::quic::test
+{
+    using namespace std::literals;
+
+    TEST_CASE("007: Server creating streams", "[007][server][streams][TOPSECRET]")
```

Comment:
Note: add test cases evaluating proper error handling for this and six. And then do that for the rest and make another PR